### PR TITLE
Fetch pool programs from onchain

### LINKED ISSUE #809: Fetch Rewards information from the Rewards contract instead of hard coding it in the frontend
Right now the rewards calculation happens in the frontend. Change this to use the smart contracts instead as described in this document:

https://docs.google.com/document/d/1KOVa-73JEdCvE6LhSWBnIbrF0ezfeaUQBtd2HdRcAe8/edit

### PR DESCRIPTION
**Pull request by H34D**
fixes #809

### CODE REVIEW

File: src/api/observables.ts [hunk 1]
```diff
@@ -115,6 +115,19 @@ export const bancorConverterRegistry$ = contractAddresses$.pipe(
...
 
+export const stakingRewards$ = contractAddresses$.pipe(
+  pluck("StakingRewards"),
+  distinctUntilChanged(compareString),
+  shareReplay(1)
+);
+
+export const poolPrograms$ = stakingRewards$.pipe(
+  switchMap(() =>
+    vxm.rewards.fetchPoolPrograms()
```
Comment (Velua) [inline src/api/observables.ts:126]: By calling ```fetchPoolPrograms``` without the param of ```stakingRewards``` are we assuming that it's going to be available through the getter in it's own method? Should we just accept a param of staking rewards here? Even optional if you wanted, that way we know it's going to be available and no side effects are necessary for this thing to run.

Comment (H34D) [reply src/api/observables.ts:126]: sure i can add that

**Comment by Velua** [comment]
@H34D is this good to go? We're just waiting on the major release before we throw this in on staging?

**Comment by H34D** [comment]
> @H34D is this good to go? We're just waiting on the major release before we throw this in on staging?

i consider this a new feature so i am waiting till the staging branch is unfrozen.

**Pull request merged**